examples/internal/example_setup/create_storage.go

```diff
@@ -0,0 +1,74 @@
+package example_setup
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/defs"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/monitor"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/services"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/storage"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk"
+
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/infra"
+)
+
+type StorageInfra struct {
+	Provider *storage.Provider
+	Monitor  *monitor.Daemon
+	Services *services.WalletServices
+}
```

Comment:
```suggestion
type StorageInfra struct {
type StorageInfra struct {
	Provider *storage.Provider
	Monitor  *monitor.Daemon
	Services *services.WalletServices
	Logger   *slog.Logger
}
```
i would add logger here